include/swift/SIL/LinearLifetimeChecker.h

```diff
@@ -148,6 +61,36 @@ class LinearLifetimeChecker {
                         DeadEndBlocks &deadEndBlocks)
       : visitedBlocks(visitedBlocks), deadEndBlocks(deadEndBlocks) {}
 
+  /// Returns true that \p value forms a linear lifetime with consuming uses \p
+  /// consumingUses, non consuming uses \p nonConsumingUses. Returns false
+  /// otherwise.
+  bool validateLifetime(SILValue value, ArrayRef<Operand *> consumingUses,
+                        ArrayRef<Operand *> nonConsumingUses);
+
+  /// Given a value and a consuming use of that value, compute a non-unique
+  /// minimal set of insertion points that together with \p consumingUse
+  /// post-dominate and end the lifetime of \p value.
+  ///
+  /// Returns true if we completed the consuming use set and discovered that \p
+  /// consumingUse is not strongly control equivalent to value (meaning
+  /// consumingUse is not in the same loop in the loop nest as value).
+  ///
+  /// \p scratch A scratch buffer to be used during our computation. If nullptr,
+  /// routine will create its own small vector.
```

Comment:
I can't tell what `scratch` is for. The clients assume it's `leakingBlocks`.

Reply:
@atrick It is just an artifact of how it is brought up. I need to thread through a visitor instead of the vector. I want to get rid of it in a forthcoming commit.